### PR TITLE
Fixes in the client-side router

### PR DESCRIPTION
**Pull request by martrapp**
## Changes

There had been some corner cases for pages that did not include `<ViewTransitions>` in there `<head>`   

## Testing

added two e2e tests

## Docs

n/a bug fix

### CODE REVIEW

**Review by natemoo-re — APPROVED** [review]
This seems totally reasonable, thanks for the fix!

File: packages/astro/components/ViewTransitions.astro [hunk 1]
```diff
@@ -25,13 +25,14 @@ const { fallback = 'animate' } = Astro.props as Props;
...
 	let currentHistoryIndex = history.state?.index || 0;
-	if (!history.state) {
+	if (!history.state && transitionEnabledOnThisPage()) {
 		persistState({ index: currentHistoryIndex, scrollY: 0 });
 	}
 
 	const supportsViewTransitions = !!document.startViewTransition;
-	const transitionEnabledOnThisPage = () =>
-		!!document.querySelector('[name="astro-view-transitions-enabled"]');
+	function transitionEnabledOnThisPage() {
```
Comment (matthewp) [inline packages/astro/components/ViewTransitions.astro:33]: went with arrow functions to save bytes, can we change this back?

Comment (martrapp) [reply packages/astro/components/ViewTransitions.astro:33]: Sure ;-)
I had the choice to move the code up, or make it a function. Will revert.

Comment (martrapp) [reply packages/astro/components/ViewTransitions.astro]: Done. This time, even the checks passed :)
